### PR TITLE
OIDC Auth: Add scopes from config, read id_token, add option for verbose logging

### PR DESCRIPTION
**Pull request by fm3**
### Steps to test:
- We tested this with a custom provider that does not return the user info in the access_token (but only in the id_token)
- Should also still work with keycloak (which sends everything everywhere)

------
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Removed dev-only changes like prints and application.conf edits

### CODE REVIEW

File: app/oxalis/security/OpenIdConnectClient.scala [hunk 1]
```diff
@@ -51,42 +55,54 @@ class OpenIdConnectClient @Inject()(rpc: RPC, conf: WkConf)(implicit ec: Executi
   Fetches token from the oidc provider (https://openid.net/specs/openid-connect-core-1_0.html#TokenRequest),
   fields described by https://www.rfc-editor.org/rfc/rfc6749#section-4.4.2
    */
-  def getAndValidateToken(redirectUrl: String, code: String): Fox[JsObject] =
+  def getAndValidateToken(redirectUrl: String, code: String): Fox[(JsObject, Option[JsObject])] =
```
Comment (frcroth) [inline app/oxalis/security/OpenIdConnectClient.scala:58]: Token**s** ?

File: app/oxalis/security/OpenIdConnectClient.scala [hunk 2]
```diff
@@ -51,42 +55,54 @@ class OpenIdConnectClient @Inject()(rpc: RPC, conf: WkConf)(implicit ec: Executi
...
+      tokenResponse: OpenIdConnectTokenResponse,
+      serverInfos: OpenIdConnectProviderInfo): Fox[(JsObject, Option[JsObject])] =
     for {
       publicKey <- fetchServerPublicKey(serverInfos)
-      decodedResponse <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
-    } yield decodedResponse
+      decdoedAccessToken <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
+      decodedIdToken: Option[JsObject] <- Fox.runOptional(tokenResponse.id_token)(itToken =>
+        JwtJson.decodeJson(itToken, publicKey).toFox)
+    } yield (decdoedAccessToken, decodedIdToken)
```
Comment (frcroth) [inline app/oxalis/security/OpenIdConnectClient.scala:99]:
```suggestion
      decodedAccessToken <- JwtJson.decodeJson(tokenResponse.access_token, publicKey).toFox
      decodedIdToken: Option[JsObject] <- Fox.runOptional(tokenResponse.id_token)(itToken =>
        JwtJson.decodeJson(itToken, publicKey).toFox)
    } yield (decodedAccessToken, decodedIdToken)
```

Comment (frcroth) [reply app/oxalis/security/OpenIdConnectClient.scala:99]: Also what does "itToken" mean here?

Comment (fm3) [reply app/oxalis/security/OpenIdConnectClient.scala]: oops, that was of course supposed to be idToken

**Review by normanrz — APPROVED** [review]
lgtm!

**Review by frcroth — APPROVED** [review]
Works with keycloak

**Pull request merged**